contracts/contracts/orand/OrandStorage.sol

```diff
@@ -1,47 +1,28 @@
-// SPDX-License-Identifier: MIT
+// SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.0;
+import '../interfaces/IOrandStorage.sol';
 
-contract OrandStorage {
-  // Storage form of proof
-  struct Epoch {
-    uint128 epoch;
-    uint64 timestamp;
-    uint64 sued;
-    uint256 y;
-    uint256[2] gamma;
-    uint256 c;
-    uint256 s;
-    address uWitness;
-    uint256[2] cGammaWitness;
-    uint256[2] sHashWitness;
-    uint256 zInv;
-  }
-
-  // Tranmission form of proof
-  struct EpochProof {
-    uint256 y;
-    uint256[2] gamma;
-    uint256 c;
-    uint256 s;
-    address uWitness;
-    uint256[2] cGammaWitness;
-    uint256[2] sHashWitness;
-    uint256 zInv;
-  }
+error InvalidEpochId();
+error SuedEpoch();
 
+contract OrandStorage is IOrandStorage {
   // Event: New Epoch
   event NewEpoch(address indexed receiverAddress, uint256 indexed epoch, uint256 indexed randomness);
 
   // Storage of epoch
   mapping(address => mapping(uint256 => Epoch)) internal storageEpoch;
 
   // Total number of epoch
-  mapping(address => uint256) totalEpoch;
+  mapping(address => uint256) internal totalEpoch;
 
   // Check if epoch is valid
   modifier onlyValidEpoch(address receiverAddress, uint256 epoch) {
-    require(epoch < totalEpoch[receiverAddress] && epoch >= 1, 'OS: Invalid epoch id');
-    require(storageEpoch[receiverAddress][epoch].sued == 0, 'OS: This epoch was sued');
+    if (epoch >= totalEpoch[receiverAddress] && epoch < 1) {
```

Comment:
```suggestion
    if (epoch >= totalEpoch[receiverAddress] || epoch < 1) {
```

Reply:
Nice catch, I was wrong when I try to convert to inverted logic